src/main/java/org/folio/innreach/domain/entity/LocalAgency.java

```diff
@@ -29,7 +33,15 @@
 @ToString(exclude = {"folioLibraryIds", "centralServer"})
 @Entity
 @Table(name = "local_agency")
+@NamedQuery(
+  name = FETCH_ONE_BY_LIBRARY_ID_NAME,
+  query = FETCH_ONE_BY_LIBRARY_ID
+)
 public class LocalAgency implements Identifiable<UUID> {
+  public static final String FETCH_ONE_BY_LIBRARY_ID_NAME = "LocalAgency.fetchOneByLibraryId";
+  public static final String FETCH_ONE_BY_LIBRARY_ID = "SELECT la FROM LocalAgency AS la " +
+    "LEFT JOIN FETCH la.centralServer " +
+    "WHERE :libraryId MEMBER OF la.folioLibraryIds";
```

Comment:
I assume there might be multiple central server configurations that have local agencies for same library. If that's the case, you should return all of them